pkg/apis/pipeline/v1/pipeline_types_test.go

```diff
@@ -835,7 +835,122 @@ func TestPipelineTask_GetMatrixCombinationsCount(t *testing.T) {
+						Name: "DOCKERFILE", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/Dockerfile2"},
+					}},
+				}, {
+					Name: "build-3",
+					Params: []Param{{
+						Name: "IMAGE", Value: ParamValue{Type: ParamTypeString, StringVal: "image-3"},
+					}, {
+						Name: "DOCKERFILE", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/Dockerfile3"},
+					}},
+				}},
+			},
+		},
+		matrixCombinationsCount: 3,
+	}, {
+		name: "params and include in matrix with overriding combinations params",
+		pt: &PipelineTask{
+			Name: "task",
+			Matrix: &Matrix{
+				Params: []Param{{
+					Name: "GOARCH", Value: ParamValue{ArrayVal: []string{"linux/amd64", "linux/ppc64le", "linux/s390x"}},
+				}, {
+					Name: "version", Value: ParamValue{ArrayVal: []string{"go1.17", "go1.18.1"}}},
+				},
+				Include: []MatrixInclude{{
+					Name: "common-package",
+					Params: []Param{{
+						Name: "package", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"}}},
+				}, {
+					Name: "s390x-no-race",
+					Params: []Param{{
+						Name: "GOARCH", Value: ParamValue{Type: ParamTypeString, StringVal: "linux/s390x"},
+					}, {
+						Name: "flags", Value: ParamValue{Type: ParamTypeString, StringVal: "-cover -v"}}},
+				}, {
+					Name: "go117-context",
+					Params: []Param{{
+						Name: "version", Value: ParamValue{Type: ParamTypeString, StringVal: "go1.17"},
+					}, {
+						Name: "context", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/go117/context"}}},
+				}},
+			}},
+		matrixCombinationsCount: 6,
+	}, {
+		name: "params and include in matrix with overriding combinations params and one new combination",
+		pt: &PipelineTask{
+			Name: "task",
+			Matrix: &Matrix{
+				Params: []Param{{
+					Name: "GOARCH", Value: ParamValue{ArrayVal: []string{"linux/amd64", "linux/ppc64le", "linux/s390x"}},
+				}, {
+					Name: "version", Value: ParamValue{ArrayVal: []string{"go1.17", "go1.18.1"}}},
+				},
+				Include: []MatrixInclude{{
+					Name: "common-package",
+					Params: []Param{{
+						Name: "package", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"}}},
+				}, {
+					Name: "s390x-no-race",
+					Params: []Param{{
+						Name: "GOARCH", Value: ParamValue{Type: ParamTypeString, StringVal: "linux/s390x"},
+					}, {
+						Name: "flags", Value: ParamValue{Type: ParamTypeString, StringVal: "-cover -v"}}},
+				}, {
+					Name: "go117-context",
+					Params: []Param{{
+						Name: "version", Value: ParamValue{Type: ParamTypeString, StringVal: "go1.17"},
+					}, {
+						Name: "context", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/go117/context"}}},
+				}, {
+					Name: "non-existent-arch",
+					Params: []Param{{
+						Name: "GOARCH", Value: ParamValue{Type: ParamTypeString, StringVal: "I-do-not-exist"}},
+					},
+				}},
+			}},
+		matrixCombinationsCount: 7,
+	}, {
+		name: "params and include in matrix with one overriding combinations params",
```

Comment:
Does this test case cover anything that the test case "params and include in matrix with overriding combinations params" doesn't?

Reply:
I removed the last test.